src/app/api/event/[_id]/role/route.ts

```diff
@@ -0,0 +1,21 @@
+import { NextRequest, NextResponse } from 'next/server'
+import connectDB from '@database/db'
+import volunteerRoleSchema from '@database/volunteerRoleSchema'
+
+type IParams = {
+    params: {
+        slug: string
+    }
+}
+
+export async function GET(req: NextRequest, { params }: IParams) {
+    await connectDB() // function from db.ts
+		const { slug } = params 
+
+	   try {
+	        const volunteerRoles = await volunteerRoleSchema.find().orFail()
+	        return NextResponse.json(volunteerRoles)
+	    } catch (err) {
+	        return NextResponse.json('No volunteer roles found.', { status: 404 })
+	    }
```

Comment:
This is incorrect, what we want is to find all the `VolunteerRoles` for the single `Event` passed in the URL, which also should be _id instead of slug. You did essentially the same thing with your DELETE endpoint that deleted all roles associated with the given Event id. Make sure to return these roles in an array in the HTTP response.